src/CodeGenerator/InjectorGenerator.php

```diff
@@ -60,25 +61,33 @@ class InjectorGenerator implements LoggerAwareInterface
      */
     private $autoloadGenerator;
 
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
     /**
      * Constructs the compiler instance
      *
      * @param ConfigInterface $config The configuration to compile from
      * @param DependencyResolverInterface $resolver The resolver to utilize
-     * @param string $namespace Namespace to use for generated class; defaults
+     * @param string|null $namespace Namespace to use for generated class; defaults
      *     to Zend\Di\Generated.
+     * @param LoggerInterface|null $logger An optional logger instance to log failures
+     *     and processed classes.
      */
     public function __construct(
         ConfigInterface $config,
         DependencyResolverInterface $resolver,
-        ?string $namespace = null
+        ?string $namespace = null,
```

Comment:
Just an FYI: you don't need to use a nullable type here when providing a default `null` value. (Something @webimpress demonstrated to me recently.)